src/cstring.rs

```diff
@@ -0,0 +1,450 @@
+use core::ffi::{c_char, CStr};
+
+/// A fixed capacity [`CString`](https://doc.rust-lang.org/std/ffi/struct.CString.html).
+///
+/// Stores up to N-1 elements with a byte reserved for the terminating nil byte.
+#[derive(Clone)]
+pub struct CString<const CAP: usize> {
```

Comment:
```suggestion
pub struct CString<const N: usize> {
```

to be consistent with the other types.